collector/src/main/java/io/prometheus/jmx/JmxScraper.java

```diff
@@ -110,12 +113,23 @@ public void doScrape() throws Exception {
                 }
             }
 
+            Map<ObjectName, HashSet<String>> attributesToExcludeFromBeans = new HashMap<ObjectName, HashSet<String>>();
+            for (Map.Entry<ObjectName, HashSet<String>> entry : excludeBeanAttributeNames.entrySet()) {
+                for (ObjectInstance instance: beanConn.queryMBeans(entry.getKey(), null)) {
+                    attributesToExcludeFromBeans.put(instance.getObjectName(), entry.getValue());
+                }
+            }
+
```

Comment:
This could cause a performance issue when using the exporter in standalone mode (and possibly agent mode.)

Add better approach would be to query the beans, then when processing through the attributes, filter out any attributes on the exclude list.